From 892a3d64076baacbf011e008eda9a43660da05e5 Mon Sep 17 00:00:00 2001
From: jeremyd2019 <github@jdrake.com>
Date: Sun, 6 Sep 2020 12:10:54 -0700
Subject: [PATCH] use rip-relative addressing on x64

If the image base is greater than 4GB, the previous method results in a linker error
tkWin32Dll.o:tkWin32Dll.c:(.text+0x4d): relocation truncated to fit: R_X86_64_32S against `.text'
---
 win/tkWin32Dll.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/win/tkWin32Dll.c b/win/tkWin32Dll.c
index 49ffd832c..9ff374e4d 100644
--- a/win/tkWin32Dll.c
+++ b/win/tkWin32Dll.c
@@ -135,7 +135,7 @@ DllMain(
 	    "leaq	%[registration], %%rdx"		"\n\t"
 	    "movq	%%gs:0,		%%rax"		"\n\t"
 	    "movq	%%rax,		0x0(%%rdx)"	"\n\t" /* link */
-	    "leaq	1f,		%%rax"		"\n\t"
+	    "leaq	1f(%%rip),	%%rax"		"\n\t"
 	    "movq	%%rax,		0x8(%%rdx)"	"\n\t" /* handler */
 	    "movq	%%rbp,		0x10(%%rdx)"	"\n\t" /* rbp */
 	    "movq	%%rsp,		0x18(%%rdx)"	"\n\t" /* rsp */
